### ISSUE TITLE
代码中的 negative_samples代表的是什么？

### ISSUE THREAD

**Issue by Macvh** [THUDM/GATNE#92]
您好，非常感谢您的开源项目
我想问问代码中negative_samples代表的是什么？
negs = torch.multinomial(
            self.sample_weights, self.num_sampled * n, replacement=True
        ).view(n, self.num_sampled)
noise = torch.neg(self.weights[negs])
sum_log_sampled = torch.sum(
    torch.log(torch.sigmoid(torch.bmm(noise, embs.unsqueeze(2)))), 1
).squeeze()
这部分代码的作用，我不太懂，这边是讲假边当作真边了吗？
希望能获得您的解答

**Comment by cenyk1230** [comment]
Hi @Macvh,

你可以查阅一下论文中的公式(17)，negs这部分对应后面那一项。

**Comment by Macvh** [comment]
好的，非常感谢您的解答
